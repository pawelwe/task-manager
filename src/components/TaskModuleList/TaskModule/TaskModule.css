.TaskModule {
  flex-basis: 100%;
  margin: 0 0 5rem;
  overflow: hidden; }
  .TaskModule_content {
    position: relative; }
  .TaskModule_title {
    font-size: 2rem;
    font-weight: bold;
    display: inline-block;
    max-width: 80%;
    word-wrap: break-word; }
  .TaskModule_filterBox {
    position: relative;
    margin: 1rem 0 2rem;
    font-size: 1.2rem; }
  .TaskModule_filters {
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    margin: 0.6rem 0; }
  .TaskModule_filterDirArrow {
    position: absolute;
    z-index: 9;
    right: 0;
    top: 0;
    font-size: 1.4rem;
    margin-right: 3px; }

@media (min-width: 768px) {
  .TaskModule {
    flex-basis: 33.333%; }
    .TaskModule_content {
      margin: 0 2rem; } }

@media (min-width: 1024px) {
  .TaskModule {
    flex-basis: 25%; }
    .TaskModule_content {
      margin: 0 3rem; } }
