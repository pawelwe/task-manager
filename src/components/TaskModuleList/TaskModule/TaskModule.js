import React from 'react';
import AddTask from './AddTask/AddTask';
import Filter from './Filter/Filter';
import TaskItemList from './TaskItemList/TaskItemList';
import classes from './TaskModule.css';

const TaskModule = ({
  title,
  tasks,
  addNewTask,
  toggleTask,
  moduleId,
  sortByProp,
  sortByExpiration,
  removeTask,
  removeTaskModule,
  sortAsc,
  sortTasksBy,
}) => {
  return (
    <li className={`${classes.TaskModule} fadeIn`}>
      <section className={classes.TaskModule_content}>
        <h4 className={classes.TaskModule_title}>{title}</h4>
        <span
          id="removeTaskModule"
          onClick={() => removeTaskModule(moduleId)}
          className="removeBtn"
        />
        <div className={classes.TaskModule_filterBox}>
          <strong>Sort by:</strong>
          <div className={classes.TaskModule_filters}>
            <Filter
              sortBy={sortByProp}
              text={'Priority'}
              filter={'priority'}
              moduleId={moduleId}
              sortTasksBy={sortTasksBy}
            />
            <Filter
              sortBy={sortByExpiration}
              text={'Expiration'}
              filter={'expiration'}
              moduleId={moduleId}
              sortTasksBy={sortTasksBy}
            />
            <Filter
              sortBy={sortByProp}
              text={'Completion'}
              filter={'completed'}
              moduleId={moduleId}
              sortTasksBy={sortTasksBy}
            />
            <Filter
              sortBy={sortByProp}
              text={'Creation'}
              filter={'creationDate'}
              moduleId={moduleId}
              sortTasksBy={sortTasksBy}
            />
          </div>
          <span className={classes.TaskModule_filterDirArrow}>
            {sortAsc ? '↑' : '↓'}
          </span>
        </div>
        <ul className={classes.TaskList}>
          {tasks &&
            tasks.length < 1 && (
              <li>
                <p>No tasks in current module...</p>
                <br />
              </li>
            )}
          <TaskItemList
            tasks={tasks}
            toggleTask={toggleTask}
            removeTask={removeTask}
            moduleId={moduleId}
          />
        </ul>
        <AddTask addNewTask={addNewTask} moduleId={moduleId} />
      </section>
    </li>
  );
};

export default TaskModule;
