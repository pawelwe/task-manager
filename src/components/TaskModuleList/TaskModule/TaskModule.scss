@import '../../App/scss/variables';

.TaskModule {
  flex-basis: 100%;
  margin: 0 0 5rem;
  overflow: hidden;

  &_content {
    position: relative;
  }

  &_title {
    font-size: 2rem;
    font-weight: bold;
    display: inline-block;
    max-width: 80%;
    word-wrap: break-word;
  }

  &_filterBox {
    position: relative;
    margin: 1rem 0 2rem;
  }

  &_filterDirArrow {
    position: absolute;
    z-index: 9;
    right: 0;
    top: 0;
    font-size: 1.4rem;
    margin-right: 3px;
  }
}

@media (min-width: $tablet) {
  .TaskModule {
    flex-basis: 33.333%;

    &_content {
      margin: 0 2rem;
    }
  }
}

@media (min-width: $desktop) {
  .TaskModule {
    flex-basis: 25%;

    &_content {
      margin: 0 3rem;
    }
  }
}
