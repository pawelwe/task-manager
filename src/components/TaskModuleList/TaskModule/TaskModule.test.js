import React from 'react';
import { shallow } from 'enzyme';
import TaskModule from './TaskModule';

describe('TaskModule Component', () => {
  const toggleModal = jest.fn();
  const wrapper = shallow(<TaskModule toggleModal={toggleModal} tasks={[]} />);

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('toggleModal called', () => {
    wrapper.find('#removeTaskModule').simulate('click');
    expect(toggleModal).toHaveBeenCalled();
  });
});
