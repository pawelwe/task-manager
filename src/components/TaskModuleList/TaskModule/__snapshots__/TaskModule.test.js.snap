// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskModule Component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskModule
    removeTaskModule={[Function]}
    tasks={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className={undefined}
>
        <h4
                className={undefined}
        />
        <span
                className="removeBtn"
                id="removeTaskModule"
                onClick={[Function]}
        />
        <div
                className={undefined}
        >
                <strong>
                        Sort by:
                </strong>
                <div
                        className={undefined}
                >
                        <Filter
                                filter="priority"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Priority"
                        />
                        <Filter
                                filter="expiration"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Expiration"
                        />
                        <Filter
                                filter="completed"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Completion"
                        />
                        <Filter
                                filter="creationDate"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Creation"
                        />
                </div>
                <span
                        className={undefined}
                >
                        ↓
                </span>
        </div>
        <ul
                className={undefined}
        >
                <li>
                        <p>
                                No tasks in current module...
                        </p>
                        <br />
                </li>
                <TaskItemList
                        moduleId={undefined}
                        removeTask={undefined}
                        tasks={Array []}
                        toggleTask={undefined}
                />
        </ul>
        <AddTask
                addNewTask={undefined}
                moduleId={undefined}
        />
</section>,
      "className": "undefined fadeIn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className={undefined}
/>,
          <span
            className="removeBtn"
            id="removeTaskModule"
            onClick={[Function]}
/>,
          <div
            className={undefined}
>
            <strong>
                        Sort by:
            </strong>
            <div
                        className={undefined}
            >
                        <Filter
                                    filter="priority"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Priority"
                        />
                        <Filter
                                    filter="expiration"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Expiration"
                        />
                        <Filter
                                    filter="completed"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Completion"
                        />
                        <Filter
                                    filter="creationDate"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Creation"
                        />
            </div>
            <span
                        className={undefined}
            >
                        ↓
            </span>
</div>,
          <ul
            className={undefined}
>
            <li>
                        <p>
                                    No tasks in current module...
                        </p>
                        <br />
            </li>
            <TaskItemList
                        moduleId={undefined}
                        removeTask={undefined}
                        tasks={Array []}
                        toggleTask={undefined}
            />
</ul>,
          <AddTask
            addNewTask={undefined}
            moduleId={undefined}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "removeBtn",
            "id": "removeTaskModule",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Sort by:
</strong>,
              <div
                className={undefined}
>
                <Filter
                                filter="priority"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Priority"
                />
                <Filter
                                filter="expiration"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Expiration"
                />
                <Filter
                                filter="completed"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Completion"
                />
                <Filter
                                filter="creationDate"
                                moduleId={undefined}
                                sortBy={undefined}
                                sortTasksBy={undefined}
                                text="Creation"
                />
</div>,
              <span
                className={undefined}
>
                ↓
</span>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by:",
              },
              "ref": null,
              "rendered": "Sort by:",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Filter
                    filter="priority"
                    moduleId={undefined}
                    sortBy={undefined}
                    sortTasksBy={undefined}
                    text="Priority"
/>,
                  <Filter
                    filter="expiration"
                    moduleId={undefined}
                    sortBy={undefined}
                    sortTasksBy={undefined}
                    text="Expiration"
/>,
                  <Filter
                    filter="completed"
                    moduleId={undefined}
                    sortBy={undefined}
                    sortTasksBy={undefined}
                    text="Completion"
/>,
                  <Filter
                    filter="creationDate"
                    moduleId={undefined}
                    sortBy={undefined}
                    sortTasksBy={undefined}
                    text="Creation"
/>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "filter": "priority",
                    "moduleId": undefined,
                    "sortBy": undefined,
                    "sortTasksBy": undefined,
                    "text": "Priority",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "filter": "expiration",
                    "moduleId": undefined,
                    "sortBy": undefined,
                    "sortTasksBy": undefined,
                    "text": "Expiration",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "filter": "completed",
                    "moduleId": undefined,
                    "sortBy": undefined,
                    "sortTasksBy": undefined,
                    "text": "Completion",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "filter": "creationDate",
                    "moduleId": undefined,
                    "sortBy": undefined,
                    "sortTasksBy": undefined,
                    "text": "Creation",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "↓",
                "className": undefined,
              },
              "ref": null,
              "rendered": "↓",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <p>
                                No tasks in current module...
                </p>
                <br />
</li>,
              <TaskItemList
                moduleId={undefined}
                removeTask={undefined}
                tasks={Array []}
                toggleTask={undefined}
/>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    No tasks in current module...
</p>,
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No tasks in current module...",
                  },
                  "ref": null,
                  "rendered": "No tasks in current module...",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "moduleId": undefined,
                "removeTask": undefined,
                "tasks": Array [],
                "toggleTask": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addNewTask": undefined,
            "moduleId": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className={undefined}
>
          <h4
                    className={undefined}
          />
          <span
                    className="removeBtn"
                    id="removeTaskModule"
                    onClick={[Function]}
          />
          <div
                    className={undefined}
          >
                    <strong>
                              Sort by:
                    </strong>
                    <div
                              className={undefined}
                    >
                              <Filter
                                        filter="priority"
                                        moduleId={undefined}
                                        sortBy={undefined}
                                        sortTasksBy={undefined}
                                        text="Priority"
                              />
                              <Filter
                                        filter="expiration"
                                        moduleId={undefined}
                                        sortBy={undefined}
                                        sortTasksBy={undefined}
                                        text="Expiration"
                              />
                              <Filter
                                        filter="completed"
                                        moduleId={undefined}
                                        sortBy={undefined}
                                        sortTasksBy={undefined}
                                        text="Completion"
                              />
                              <Filter
                                        filter="creationDate"
                                        moduleId={undefined}
                                        sortBy={undefined}
                                        sortTasksBy={undefined}
                                        text="Creation"
                              />
                    </div>
                    <span
                              className={undefined}
                    >
                              ↓
                    </span>
          </div>
          <ul
                    className={undefined}
          >
                    <li>
                              <p>
                                        No tasks in current module...
                              </p>
                              <br />
                    </li>
                    <TaskItemList
                              moduleId={undefined}
                              removeTask={undefined}
                              tasks={Array []}
                              toggleTask={undefined}
                    />
          </ul>
          <AddTask
                    addNewTask={undefined}
                    moduleId={undefined}
          />
</section>,
        "className": "undefined fadeIn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className={undefined}
/>,
            <span
              className="removeBtn"
              id="removeTaskModule"
              onClick={[Function]}
/>,
            <div
              className={undefined}
>
              <strong>
                            Sort by:
              </strong>
              <div
                            className={undefined}
              >
                            <Filter
                                          filter="priority"
                                          moduleId={undefined}
                                          sortBy={undefined}
                                          sortTasksBy={undefined}
                                          text="Priority"
                            />
                            <Filter
                                          filter="expiration"
                                          moduleId={undefined}
                                          sortBy={undefined}
                                          sortTasksBy={undefined}
                                          text="Expiration"
                            />
                            <Filter
                                          filter="completed"
                                          moduleId={undefined}
                                          sortBy={undefined}
                                          sortTasksBy={undefined}
                                          text="Completion"
                            />
                            <Filter
                                          filter="creationDate"
                                          moduleId={undefined}
                                          sortBy={undefined}
                                          sortTasksBy={undefined}
                                          text="Creation"
                            />
              </div>
              <span
                            className={undefined}
              >
                            ↓
              </span>
</div>,
            <ul
              className={undefined}
>
              <li>
                            <p>
                                          No tasks in current module...
                            </p>
                            <br />
              </li>
              <TaskItemList
                            moduleId={undefined}
                            removeTask={undefined}
                            tasks={Array []}
                            toggleTask={undefined}
              />
</ul>,
            <AddTask
              addNewTask={undefined}
              moduleId={undefined}
/>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "removeBtn",
              "id": "removeTaskModule",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Sort by:
</strong>,
                <div
                  className={undefined}
>
                  <Filter
                                    filter="priority"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Priority"
                  />
                  <Filter
                                    filter="expiration"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Expiration"
                  />
                  <Filter
                                    filter="completed"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Completion"
                  />
                  <Filter
                                    filter="creationDate"
                                    moduleId={undefined}
                                    sortBy={undefined}
                                    sortTasksBy={undefined}
                                    text="Creation"
                  />
</div>,
                <span
                  className={undefined}
>
                  ↓
</span>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort by:",
                },
                "ref": null,
                "rendered": "Sort by:",
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Filter
                      filter="priority"
                      moduleId={undefined}
                      sortBy={undefined}
                      sortTasksBy={undefined}
                      text="Priority"
/>,
                    <Filter
                      filter="expiration"
                      moduleId={undefined}
                      sortBy={undefined}
                      sortTasksBy={undefined}
                      text="Expiration"
/>,
                    <Filter
                      filter="completed"
                      moduleId={undefined}
                      sortBy={undefined}
                      sortTasksBy={undefined}
                      text="Completion"
/>,
                    <Filter
                      filter="creationDate"
                      moduleId={undefined}
                      sortBy={undefined}
                      sortTasksBy={undefined}
                      text="Creation"
/>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "filter": "priority",
                      "moduleId": undefined,
                      "sortBy": undefined,
                      "sortTasksBy": undefined,
                      "text": "Priority",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "filter": "expiration",
                      "moduleId": undefined,
                      "sortBy": undefined,
                      "sortTasksBy": undefined,
                      "text": "Expiration",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "filter": "completed",
                      "moduleId": undefined,
                      "sortBy": undefined,
                      "sortTasksBy": undefined,
                      "text": "Completion",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "filter": "creationDate",
                      "moduleId": undefined,
                      "sortBy": undefined,
                      "sortTasksBy": undefined,
                      "text": "Creation",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "↓",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "↓",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <p>
                                    No tasks in current module...
                  </p>
                  <br />
</li>,
                <TaskItemList
                  moduleId={undefined}
                  removeTask={undefined}
                  tasks={Array []}
                  toggleTask={undefined}
/>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      No tasks in current module...
</p>,
                    <br />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No tasks in current module...",
                    },
                    "ref": null,
                    "rendered": "No tasks in current module...",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "moduleId": undefined,
                  "removeTask": undefined,
                  "tasks": Array [],
                  "toggleTask": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addNewTask": undefined,
              "moduleId": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
